### PR TITLE
basic tree context

### PR DESCRIPTION
**Pull request by Cats-n-coffee**
Basic tree context with a provider and state

### CODE REVIEW

File: src/context/TreeContext.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+
+interface TreeNode {
+  name: string;
+  props: Array<PropFormValue>;
+  state: Array<StateFormValue>;
+  parent: string;
+  children?: Array<TreeNode>;
```
Comment (katesroad) [inline src/context/TreeContext.tsx]: `TreeNode[]`

File: src/context/TreeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  const value = { tree };
+
+  return <TreeContext.Provider value={value}>{children}</TreeContext.Provider>;
+}
+
+export function useTreeContext() {
+  const tree = React.useContext(TreeContext);
+
+  if (!tree) {
```
Comment (katesroad) [inline src/context/TreeContext.tsx:55]: we have a `treeContextDefault` , so the context value won't be undefined.

**Review by katesroad — CHANGES_REQUESTED** [review]

File: src/context/TreeContext.tsx (hunk 1 above)
Comment (Cats-n-coffee) [reply src/context/TreeContext.tsx]: The code has changed on those lines, but I still modified with what you suggested on the spots where TreeNode[] is used.

File: src/context/TreeContext.tsx (hunk 2 above)
Comment (Cats-n-coffee) [reply src/context/TreeContext.tsx:55]: I see what you mean, but if the context hook is requested outside of its provider, won't it be undefined?

Comment (katesroad) [reply src/context/TreeContext.tsx:55]: No

Comment (Cats-n-coffee) [reply src/context/TreeContext.tsx:55]: Okay I found a solution. Default value is undefined now, so this line will work.

**Review by katesroad — APPROVED** [review]

**Pull request merged**